pypit/core/artracewave.py

```diff
@@ -0,0 +1,1235 @@
+            trcdict = pad_dict(trcdict)
+            continue
+        for k in range(sz+1-nsmth, sz+1):
+            xtfit[k+sz] = ordcen[arcdet[j], slitnum]+k
+        # Fit down
+        pcen = arcdet[j]
+        centv = None
+        for k in range(1,sz+1-nsmth):
+            if (pcen < nspecfit) or (pcen > msarc.shape[0]-(nspecfit+1)):
+                offchip = True
+                break
+            if ordcen[pcen, slitnum]-k < 0:
+                offchip = True
+                break
+            # yfit = msarc[pcen-nspecfit:pcen+nspecfit+1,ordcen[arcdet[j],0]-k]
+            yfit = msarc[pcen - nspecfit:pcen + nspecfit + 1,
+                   ordcen[arcdet[j], slitnum] - k - nsmth:ordcen[arcdet[j], slitnum] - k + nsmth + 1]
+            if len(yfit.shape) == 2:
+                yfit = np.median(yfit, axis=1)
+            if np.size(yfit) == 0:
+                offchip = True
+                break
+            wgd = np.where(yfit == maskval)
+            if wgd[0].size != 0:
+                continue
+            if method == "fweight":
+                if centv is None:
+                    centv = np.sum(yfit * (pcen+xfit))/np.sum(yfit)
+                wfit[0] = 0.5 + (pcen-centv)
+                wfit[-1] = 0.5 - (pcen-centv)
+                sumxw = yfit * (pcen+xfit) * wfit
+                sumw = yfit * wfit
+                centv = np.sum(sumxw)/np.sum(sumw)
+                fail = False
+            elif method == "cc":
+                # Get a copy of the array that will be used to cross-correlate
+                # (testcc is probably already False from the Fit Up part of the loop, but it's best to be sure)
+                if tstcc:
+                    # ccyfit = msarc[arcdet[j]-nspecfit:arcdet[j]+nspecfit+1,ordcen[arcdet[j],0]-k]
+                    ccyfit = msarc[arcdet[j]-nspecfit:arcdet[j]+nspecfit+1,ordcen[arcdet[j],slitnum]-k-nsmth:ordcen[arcdet[j],slitnum]-k+nsmth+1]
+                    if len(ccyfit.shape) == 2:
+                        ccyfit = np.median(ccyfit, axis=1)
+                    wgd = np.where(ccyfit == maskval)
+                    if wgd[0].size != 0:
+                        continue
+                    ccval = arcdet[j] + np.sum(xfit*ccyfit)/np.sum(ccyfit)
+                    tstcc = False  # Once we have an array, there's no need to keep looking
+                cc = np.correlate(ccyfit, yfit, mode='same')
+                params, fail = arutils.gauss_lsqfit(xfit, cc, 0.0)
+                centv = ccval + pcen - arcdet[j] - params[1]
+            xtfit[sz-k] = ordcen[arcdet[j], slitnum] - k
+            ytfit[sz-k] = centv
+            etfit[sz-k] = 0.02
+            #apfit[sz-k] = params[0]
+            if fail:
+                mtfit[sz-k] = 1
+            else:
+                #from IPython import embed
+                if np.isfinite(centv) is False: debugger.set_trace() #embed()
+                pcen = int(0.5 + centv)
+                mtfit[sz-k] = 0
+        '''
```

Comment:
Did you intend for this code to be deleted now?

Reply:
removed